### PR TITLE
Fixes #13452 - Support download_policy for importer

### PR DESCRIPTION
**Pull request by daviddavis**

### CODE REVIEW

File: lib/runcible/models/yum_importer.rb [hunk 1]
```diff
@@ -3,6 +3,13 @@ module Models
     class YumImporter < Importer
       ID = 'yum_importer'
       REPO_TYPE = 'rpm-repo'
+      DOWNLOAD_POLICIES = [
+        DOWNLOAD_IMMEDIATE = 'immediate',
+        DOWNLOAD_ON_DEMAND = 'on_demand',
+        DOWNLOAD_BACKGROUND = 'background'
+      ].freeze
```
Comment (ehelms) [inline lib/runcible/models/yum_importer.rb]: Why does this need to be explicitly frozen?

File: lib/runcible/models/yum_importer.rb [hunk 2]
```diff
@@ -3,6 +3,13 @@ module Models
     class YumImporter < Importer
       ID = 'yum_importer'
       REPO_TYPE = 'rpm-repo'
+      DOWNLOAD_POLICIES = [
+        DOWNLOAD_IMMEDIATE = 'immediate',
+        DOWNLOAD_ON_DEMAND = 'on_demand',
+        DOWNLOAD_BACKGROUND = 'background'
```
Comment (ehelms) [inline lib/runcible/models/yum_importer.rb]: The `DOWNLOAD_` feels a be redundant

**Comment by ehelms** [comment]
I take it this download policy is only valid for yum repositories within Pulp?

File: lib/runcible/models/yum_importer.rb (hunk 2 above)
Comment (daviddavis) [inline lib/runcible/models/yum_importer.rb]: `YumImporter::IMMEDIATE`, YumImporter::ON_DEMAND`, etc feels a bit weird to me.

File: lib/runcible/models/yum_importer.rb (hunk 1 above)
Comment (daviddavis) [inline lib/runcible/models/yum_importer.rb]: Ruby doesn't warn you if you modify a constant that's an array.

**Comment by daviddavis** [comment]
@ehelms yes, that's correct (for pulp 2.8 at least). There may be support for other content types in the future.

File: lib/runcible/models/yum_importer.rb (hunk 2 above)
Comment (ehelms) [inline lib/runcible/models/yum_importer.rb]: Wouldn't it be YumImporter::DOWNLOAD_POLICIES::DOWNLOAD_IMMEDIATE ? Or is this code chunk just confusing me all together about what it is actually doing?

Comment (daviddavis) [inline lib/runcible/models/yum_importer.rb]: I think so. What I have is equivalent to:

```ruby
DOWNLOAD_IMMEDIATE = 'immediate'
DOWNLOAD_ON_DEMAND = 'on_demand'
DOWNLOAD_BACKGROUND = 'background'
DOWNLOAD_POLICIES = [DOWNLOAD_IMMEDIATE, DOWNLOAD_ON_DEMAND, DOWNLOAD_BACKGROUND].freeze
```
I can change it to this if you prefer.

Comment (ehelms) [inline lib/runcible/models/yum_importer.rb]: I think that would help me (so i assume others?) not get stuck on trying to interpret this chunk of code to change it :)

**Comment by ehelms** [comment]
ACK

**Pull request merged**